ql/src/java/org/apache/hadoop/hive/ql/Compiler.java

```diff
@@ -265,7 +266,15 @@ private void openTransaction(TxnType txnType) throws LockException, CommandProce
     if (DriverUtils.checkConcurrency(driverContext) && startImplicitTxn(driverContext.getTxnManager()) &&
         !driverContext.getTxnManager().isTxnOpen() && txnType != TxnType.COMPACTION) {
       String userFromUGI = DriverUtils.getUserFromUGI(driverContext);
-      driverContext.getTxnManager().openTxn(context, userFromUGI, txnType);
+      boolean isHiveReplTxn = (driverContext.getQueryState().getHiveOperation() != null)
```

Comment:
(driverContext.getQueryState().getHiveOperation() != null) check is not needed

Reply:
For some of transactions such as Update_table, there is no corresponding HiveOperation (null in that case). So, it would give NPE if we compare its value.

Reply:
Done